src/main/java/application/IncompleteSessionController.java

```diff
@@ -0,0 +1,115 @@
+import java.util.concurrent.locks.Lock;
+
+/**
+ * Controller class (API endpoint) containing all incomplete session management commands
+ *
+ */
+@Api(value = "Incomplete Session Controller", description = "Operations for navigating managing incomplete sessions")
+@RestController
+@EnableAutoConfiguration
+public class IncompleteSessionController extends AbstractBaseController{
+
+    @Value("${commcarehq.host}")
+    private String host;
+
+    @Autowired
+    protected FormSessionRepo migratedFormSessionRepo;
+
+    private final Log log = LogFactory.getLog(IncompleteSessionController.class);
+
+    @ApiOperation(value = "Open an incomplete form session")
+    @RequestMapping(value = Constants.URL_INCOMPLETE_SESSION , method = RequestMethod.POST)
+    public NewFormResponse openIncompleteForm(@RequestBody IncompleteSessionRequestBean incompleteSessionRequestBean,
+                                              @CookieValue(Constants.POSTGRES_DJANGO_SESSION_ID) String authToken) throws Exception {
+        log.info("Incomplete session request with bean: " + incompleteSessionRequestBean + " sessionId :" + authToken);
+        SerializableFormSession session;
+        restoreFactory.configure(incompleteSessionRequestBean, new DjangoAuth(authToken));
+        try {
+            session = formSessionRepo.findOneWrapped(incompleteSessionRequestBean.getSessionId());
+        } catch(FormNotFoundException e) {
+            session = migratedFormSessionRepo.findOneWrapped(incompleteSessionRequestBean.getSessionId());
+            // Move over to formplayer db
+            formSessionRepo.save(session);
+        }
+        Lock lock = getLockAndBlock(session.getUsername());
+        try {
+            NewFormResponse response = newFormResponseFactory.getResponse(session);
+            log.info("Return incomplete session response: " + response);
+            return response;
+        } finally {
+            lock.unlock();
+        }
+    }
+    @ApiOperation(value = "Get a list of the current user's sessions")
+    @RequestMapping(value = Constants.URL_GET_SESSIONS, method = RequestMethod.POST)
+    public GetSessionsResponse getSessions(@RequestBody GetSessionsBean getSessionRequest,
+                                           @CookieValue(Constants.POSTGRES_DJANGO_SESSION_ID) String authToken) throws Exception {
+        log.info("Get Session Request: " + getSessionRequest);
+
+        String username = TableBuilder.scrubName(getSessionRequest.getUsername());
+
+        restoreFactory.configure(getSessionRequest, new DjangoAuth(authToken));
+
+        List<SerializableFormSession> migratedSessions = migratedFormSessionRepo.findUserSessions(
+                getSessionRequest.getUsername());
+
+        List<SerializableFormSession> formplayerSessions = formSessionRepo.findUserSessions(username);
+
+        ArrayList<FormSession> formSessions = new ArrayList<>();
+        restoreFactory.configure(getSessionRequest, new DjangoAuth(authToken));
```

Comment:
it may be nice to abstract this out so we can 1. remove the migration code easily and 2. run it in a separate command. i imagine that once we switch the toggle on for everyone and are happy with it, we'll want to just migrate all form sessions and then delete any migration code